### DISCUSSION TITLE
No cl.e binary found after installing IRAF

### DISCUSSION THREAD

**Discussion by hector500** [iraf-community/iraf#191] [Q&A] [answered]
Good day,
I'm new to linux and IRAF and noticed another user had the exact same question. Unfortunately, their response was not detailed enough to help me fix the same problem after installing IRAF on Ubuntu 20.04. I updated my system and followed all the steps on https://iraf-community.github.io/ to install the program. After a clean installation and doing [cd IRAF] (my IRAF login directory)  and [mkiraf], a [login.cl] and a [login.cl.OLD] files were created indeed, and also the [uparm] directory. The problem is that IRAF was not launched after doing [cl] and reported the [ERROR:  No /iraf/iraf/iraf-2.16.1-2021.06.14//bin.linux64/cl.e binary found]. 

I did [allocate cl.e] and the [cl.e] and [ecl.e] files are found here: 
/usr/lib/iraf/bin/cl.e
/usr/lib/iraf/bin/ecl.e

My local bin directory is /usr/local/bin

Could you please help me understand if this is a problem with the path to the IRAF binaries and how to fix it ?  Another user suggested changing the path to something including [/iraf/iraf] where my source IRAF code is also allocated. I attach here three snapshots that show the installation process and the cl.e error. Thank you, I've been struggling with this for the last two days. 

![IRAF01](https://user-images.githubusercontent.com/90478663/132913194-bbaaace4-2103-4fb1-a06f-b72ca18c4ad0.jpg)
![IRAF02](https://user-images.githubusercontent.com/90478663/132913209-415eb218-0c4d-4046-a0d9-aaf50f78cc3f.jpg)
![IRAF03](https://user-images.githubusercontent.com/90478663/132913215-4891188a-e65d-42e8-a999-77a908329cd4.jpg)

**Comment by olebole** [comment, marked as answer]
On Ubuntu 20.04, you can install IRAF directly from the repository with the command
```shell
sudo apt install iraf
```
This will give you a command [`irafcl`](http://manpages.ubuntu.com/manpages/focal/en/man1/irafcl.1.html) which starts IRAF (mkiraf is not needed there, as irafcl does the necessary setup if needed). You need to remove the settings from your previous steps. That should be the easiest way and is recommended. And since you found the files in `/usr/lib/iraf/`, you probably already installed Linux from Ubuntu.

If you really want to install from source, you need to do the following, as described in https://iraf-community.github.io/install.html:

```bash
./install
make linux64
make sysgen 2>&1 | tee build.log
```
From your screenshots, you forgot the second and third step.

**Reply by hector500** [reply]
Olebole, thanks a lot !  Your solution worked perfectly. I remember trying "make linux64" after doing "./install", but either I was in the wrong directory (not in /iraf/iraf/iraf-2.16.1-2021.06.14, my IRAF source directory) or I didn't use the "sudo" command. The point is "make linux64" did not work at first. I also made the mistake of not trying these last two steps again because I incorrectly assumed they were somehow "optional". They are actually part of the installation process. 

I will try the "sudo apt install iraf" option on another Ubuntu system just to verify how efficient the process is. 

My very best wishes,
hector500
